src/transformers/pipelines/visual_question_answering.py

```diff
@@ -55,15 +60,28 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.check_model_type(MODEL_FOR_VISUAL_QUESTION_ANSWERING_MAPPING)
 
-    def _sanitize_parameters(self, top_k=None, padding=None, truncation=None, **kwargs):
-        preprocess_params, postprocess_params = {}, {}
+        if self.model.config.__class__.__name__ == "GitConfig":
+            self.model_type = ModelType.GIT
+        elif self.model.config.__class__.__name__ in ["ViltConfig"]:
+            self.model_type = ModelType.CLASSIFIER
+
+    def _sanitize_parameters(
```

Comment:
The `_sanitize_parameters` method can look a lot like the one of the text generation pipeline: https://github.com/huggingface/transformers/blob/3658488ff77ff8d45101293e749263acf437f4d5/src/transformers/pipelines/text_generation.py#L90. No need to include things like `num_beams` here